statsd/test_helpers_test.go

```diff
@@ -0,0 +1,411 @@
+				fmt.Sprintf("datadog.dogstatsd.client.aggregated_context_by_type:%d|c%s,metrics_type:distribution", ts.telemetry.aggregated_distribution, tags),
+				fmt.Sprintf("datadog.dogstatsd.client.aggregated_context_by_type:%d|c%s,metrics_type:histogram", ts.telemetry.aggregated_histogram, tags),
+				fmt.Sprintf("datadog.dogstatsd.client.aggregated_context_by_type:%d|c%s,metrics_type:timing", ts.telemetry.aggregated_timing, tags),
+			}...)
+		}
+	}
+	return metrics
+}
+
+// Default testing scenarios
+
+func (ts *testServer) getFinalTags(t ...string) string {
+	if t == nil && ts.tags == "" {
+		return ""
+	}
+
+	res := "|#"
+	if ts.tags != "" {
+		res += ts.tags
+	}
+
+	if t != nil {
+		if ts.tags != "" {
+			res += ","
+		}
+		res += strings.Join(t, ",")
+	}
+	return res
+}
+
+func (ts *testServer) getFinalTelemetryTags() string {
+	base := "|#"
+	if ts.tags != "" {
+		base += ts.tags + ","
+	}
+	return base + strings.Join(
+		[]string{clientTelemetryTag, clientVersionTelemetryTag, "client_transport:" + ts.proto},
+		",")
+}
+
+func (ts *testServer) sendAllMetrics(c *Client) []string {
+	tags := []string{"custom:1", "custom:2"}
+	c.Gauge("Gauge", 1, tags, 1)
+	c.Count("Count", 2, tags, 1)
+	c.Histogram("Histogram", 3, tags, 1)
+	c.Distribution("Distribution", 4, tags, 1)
+	c.Decr("Decr", tags, 1)
+	c.Incr("Incr", tags, 1)
+	c.Set("Set", "value", tags, 1)
+	c.Timing("Timing", 5*time.Second, tags, 1)
+	c.TimeInMilliseconds("TimeInMilliseconds", 6, tags, 1)
+
+	ts.telemetry.gauge += 1
+	ts.telemetry.histogram += 1
+	ts.telemetry.distribution += 1
+	ts.telemetry.count += 3
+	ts.telemetry.set += 1
+	ts.telemetry.timing += 2
+
+	if ts.aggregation {
+		ts.telemetry.aggregated_context += 5
+		ts.telemetry.aggregated_gauge += 1
+		ts.telemetry.aggregated_count += 3
+		ts.telemetry.aggregated_set += 1
+	}
+	if ts.extendedAggregation {
+		ts.telemetry.aggregated_context += 4
+		ts.telemetry.aggregated_histogram += 1
+		ts.telemetry.aggregated_distribution += 1
+		ts.telemetry.aggregated_timing += 2
+	}
+
+	finalTags := ts.getFinalTags(tags...)
+
+	return []string{
+		ts.namespace + "Gauge:1|g" + finalTags,
+		ts.namespace + "Count:2|c" + finalTags,
+		ts.namespace + "Histogram:3|h" + finalTags,
+		ts.namespace + "Distribution:4|d" + finalTags,
+		ts.namespace + "Decr:-1|c" + finalTags,
+		ts.namespace + "Incr:1|c" + finalTags,
+		ts.namespace + "Set:value|s" + finalTags,
+		ts.namespace + "Timing:5000.000000|ms" + finalTags,
+		ts.namespace + "TimeInMilliseconds:6.000000|ms" + finalTags,
+	}
+}
+
+func (ts *testServer) sendAllType(c *Client) []string {
+	res := ts.sendAllMetrics(c)
+	c.SimpleEvent("hello", "world")
+	c.SimpleServiceCheck("hello", Warn)
+
+	ts.telemetry.event += 1
+	ts.telemetry.service_check += 1
+
+	finalTags := ts.getFinalTags()
+
+	return append(
+		res,
+		"_e{5,5}:hello|world"+finalTags,
+		"_sc|hello|1"+finalTags,
+	)
+}
+
+func (ts *testServer) sendBasicAggregationMetrics(client *Client) []string {
```

Comment:
seems to be unused

Reply:
True, but it will when we use the new test helper everywhere. We have a lot of duplicate code in our test that the new helpers would remove. I didn't want this PR to get out of hand. But I plan an upgrading the test suite with the new test helper soon.